### PR TITLE
docs: Add session affinity to kubeproxy-free guide

### PR DESCRIPTION
**Pull request by brb**

### CODE REVIEW

**Review by tgraf — APPROVED** [review]

File: Documentation/gettingstarted/kubeproxy-free.rst [hunk 1]
```diff
@@ -598,6 +598,31 @@ The current Cilium kube-proxy replacement mode can also be introspected through
...
+Each connection from the same pod or host to a service configured with
+``sessionAffinity: ClientIP`` will always select the same service endpoint.
+The default timeout for the affinity is three hours (updated by each request to
+the service), but it can be configured through Kubernetes' ``sessionAffinityConfig``
+if needed.
+
+The source for the affinity depends on the origin of a request. If a request is
+sent from outside the cluster to the service, the request's source IP address is
+used for determining the endpoint affinity. If a request is sent from inside
+the cluster - a network namespace cookie of a client. The latter was introduced
```
Comment (pchaigno) [inline Documentation/gettingstarted/kubeproxy-free.rst:614]:
```suggestion
used for determining the endpoint affinity. If a request is sent from inside
the cluster, the client's network namespace cookie is used. The latter was
introduced
```

**Review by borkmann — APPROVED** [review]

File: Documentation/gettingstarted/kubeproxy-free.rst [hunk 2]
```diff
@@ -598,6 +598,31 @@ The current Cilium kube-proxy replacement mode can also be introspected through
...
+in the 5.7 Linux kernel to implement the affinity at the socket layer at which
+:ref:`host-services` operates (a source IP is not available there, as the endpoint
+selection happens before a network packet has been built by kernel).
```
Comment (pchaigno) [inline Documentation/gettingstarted/kubeproxy-free.rst:617]:
```suggestion
selection happens before a network packet has been built by the kernel).
```

File: Documentation/gettingstarted/kubeproxy-free.rst [hunk 3]
```diff
@@ -598,6 +598,31 @@ The current Cilium kube-proxy replacement mode can also be introspected through
...
+The default timeout for the affinity is three hours (updated by each request to
+the service), but it can be configured through Kubernetes' ``sessionAffinityConfig``
+if needed.
+
+The source for the affinity depends on the origin of a request. If a request is
+sent from outside the cluster to the service, the request's source IP address is
+used for determining the endpoint affinity. If a request is sent from inside
+the cluster - a network namespace cookie of a client. The latter was introduced
+in the 5.7 Linux kernel to implement the affinity at the socket layer at which
+:ref:`host-services` operates (a source IP is not available there, as the endpoint
```
Comment (pchaigno) [inline Documentation/gettingstarted/kubeproxy-free.rst:616]: The title for `host-services` is `Host-Reachable Services`, so:
```suggestion
:ref:`host-services` operate (a source IP is not available there, as the endpoint
```

**Review by pchaigno — COMMENTED** [review]
A few nits on syntax and grammar, but looks like I wasn't quick enough :cry:

**Pull request merged**